app/plugin_api.go

```diff
@@ -326,6 +326,10 @@ func (api *PluginAPI) GetChannel(channelId string) (*model.Channel, *model.AppEr
 	return api.app.GetChannel(channelId)
 }
 
+func (api *PluginAPI) GetChannels(options *model.GetChannelsOptions) (*model.ChannelList, *model.AppError) {
```

Comment:
I'm wondering if it makes sense to move `Page` and `PerPage` from the options struct to a parameter. This way it's clear, that these two parameters are required.


```suggestion
func (api *PluginAPI) GetChannels(page, perPage int, options *model.GetChannelsOptions) (*model.ChannelList, *model.AppError) {
```

Reply:
There is already a convention, I guess, to have `page` and `perPage` fileds in the options struct. We can view them as optional.

Reply:
@hanzei @iomodo decided to keep the `page` and `pagePage` field in the options struct

Reply:
@iomodo and I synced about this offline and we like to have `page` and `perPage` as arguments. This makes it very clear that they are needed. Would you please changed this @dnguy078?

Reply:
Just for additional context, @dnguy078, the thinking here is to structure our RPC API as:
```
FuncName(required1, required2, options)
```

Any field that's safe to leave at its `nil` value can stay in `options`, but required fields should be explicit. We consider `page` and `perPage` required, since a `perPage` of `0`  will yield no results, which usually confuses folks, and while a `page` of `0` is valid, failing to design your API calls to account for paging could yield unexpected results if you don't realize there are more pages to view.